### PR TITLE
Feature/refactor task object

### PR DESCRIPTION
**Pull request by thetnaingtta**

### CODE REVIEW

File: src/Task/domain/entity/Task.spec.ts [hunk 1]
```diff
@@ -1,14 +1,25 @@
...
       },
+      category: 'Sample Category',
+      location: 'Sample Location',
+      budget: 100,
+      scheduling: SchedulingOption.Flexible,
+      timeslot,
+      createdon: new Date(),
     });
 
     if (task.taskIsValid && typeof task.taskIsValid === 'function') {
```
Comment (sithwin) [inline src/Task/domain/entity/Task.spec.ts:25]: it is not the practice to put if else at unit test. test should also follow SRP

File: src/Task/domain/entity/Task.spec.ts [hunk 2]
```diff
@@ -18,60 +29,222 @@ describe('Task Entity', () => {
...
         fullName: 'John Doe',
       },
+      category: 'Sample Category',
+      location: 'Sample Location',
+      budget: 100,
+      scheduling: SchedulingOption.Flexible,
+      createdon: new Date(),
     });
 
     if (task.taskIsValid && typeof task.taskIsValid === 'function') {
```
Comment (sithwin) [inline src/Task/domain/entity/Task.spec.ts:47]: better to create 2 unit tests instead of if else

File: src/Task/domain/entity/Task.spec.ts [hunk 3]
```diff
@@ -18,60 +29,222 @@ describe('Task Entity', () => {
...
       },
+      category: 'Sample Category',
+      location: 'Sample Location',
+      budget: 100,
+      scheduling: SchedulingOption.SpecificDate,
+      specificDate,
+      createdon: new Date(),
+    });
+
+    if (task.taskIsValid && typeof task.taskIsValid === 'function') {
```
Comment (sithwin) [inline src/Task/domain/entity/Task.spec.ts:74]: same here

File: src/Task/domain/entity/TaskInterface.ts [hunk 4]
```diff
@@ -2,8 +2,28 @@ export interface TaskInterface {
...
+  location?: string;
+  budget: number;
+  scheduling: SchedulingOption;
+  specificDate?: Date;
+  timeslot?: TimeSlot;
+  createdon?: Date;
+}
+
+export enum SchedulingOption {
+  Flexible = 'flexible',
```
Comment (sithwin) [inline src/Task/domain/entity/TaskInterface.ts:16]: enum usually use ALL_CAPS_WITH_UNDERSCORE

**Review by sithwin — CHANGES_REQUESTED** [review]
Amazing Job!!!

File: src/Task/domain/entity/Task.spec.ts (hunk 1 above)
Comment (thetnaingtta) [reply src/Task/domain/entity/Task.spec.ts:25]: fixed it, please take a look.

File: src/Task/domain/entity/Task.spec.ts (hunk 2 above)
Comment (thetnaingtta) [reply src/Task/domain/entity/Task.spec.ts:47]: fixed it, please take a look.

File: src/Task/domain/entity/TaskInterface.ts (hunk 4 above)
Comment (thetnaingtta) [reply src/Task/domain/entity/TaskInterface.ts:16]: fixed it, please take a look.

**Comment by thetnaingtta** [comment]
fixed and created new PR

**Pull request closed without merging**